lib/permissions/proposals/listProposalCategoryPermissions.ts

```
@@ -0,0 +1,64 @@
+import { prisma } from 'db';
+import { ProposalCategoryNotFoundError } from 'lib/proposal/errors';
+import { hasAccessToSpace } from 'lib/users/hasAccessToSpace';
+import { InvalidInputError } from 'lib/utilities/errors';
+import { isUUID } from 'lib/utilities/strings';
+
+import type { PermissionCompute } from '../interfaces';
+
+import type { AssignedProposalCategoryPermission } from './interfaces';
+import { mapProposalCategoryPermissionToAssignee } from './mapProposalCategoryPermissionToAssignee';
+
+export async function listProposalCategoryPermissions({
+  resourceId,
+  userId
+}: PermissionCompute): Promise<AssignedProposalCategoryPermission[]> {
+  if (!userId) {
+    return [];
+  }
+
+  if (!isUUID(resourceId)) {
+    throw new InvalidInputError('Invalid proposal category ID');
+  }
+
+  const proposalCategory = await prisma.proposalCategory.findUnique({
+    where: {
+      id: resourceId
+    },
+    select: {
+      spaceId: true
+    }
+  });
+
+  if (!proposalCategory) {
+    throw new ProposalCategoryNotFoundError(resourceId);
+  }
+
+  const { error } = await hasAccessToSpace({
+    spaceId: proposalCategory.spaceId,
+    userId
+  });
+
+  if (error) {
+    return [];
```

we should probably throw an error here instead